proposal/forward-with.md

```diff
@@ -105,66 +113,50 @@ $hue: 0 !default;
 Multiple configurations can be chained in a single cascading "thread" that
 contains zero or more `@forward` rules, and zero or one terminal `@use` rule.
 Variables remain open to configuration in the chain as long as every mention
-includes the `!default` flag:
-
-```scss
-// _origin.scss
-$hue: 0 !default;
-
-// _middle1.scss
-@forward "origin" with (
-  $hue: 330 !default
-);
-
-// _middle2.scss
-@forward "middle1" with (
-  $hue: 120 !default
-);
-
-// entrypoint.scss
-@use "middle2" with (
-  $hue: 30
-);
-
-// middle2.$hue == 30
-```
-
-Multiple threads configuring a single module will cause an error, no matter
-what combinations of `@forward` and `@use` are involved:
-
-```scss
-// _origin.scss
-$hue: 0 !default;
-
-// _middle1.scss
-@forward "origin" with (
-  $hue: 330 !default
-);
-
-// _middle2.scss
-@use "origin" with (
-  $hue: 120
-);
-
-// entrypoint.scss
-@forward "middle1";
-@use "middle2"; // ERROR
-```
+includes the `!default` flag. Multiple threads configuring a single module will
+cause an error, even if they originate in the same file.
 
 ## Syntax
 
 The new `WithClause` extends `@forward` to the follow grammar:
 
 <x><pre>
 **ForwardRule**     ::= '@forward' QuotedString AsClause? (ShowClause | HideClause)?  WithClause?
-**AsClause**        ::= 'as' Identifier '*'
-**ShowClause**      ::= 'show' MemberName (',' MemberName)*
-**HideClause**      ::= 'hide' MemberName (',' MemberName)*
-**MemberName**      ::= '$'? Identifier
 **WithClause**      ::= 'with' '('
 &#32;                     KeywordArgument (',' KeywordArgument)\* ','?
 &#32;                   ')'
 **KeywordArgument** ::= '$' Identifier ':' Expression ('!default')
 </pre></x>
 
 ## Semantics
+
+The `@forward ... with` semantics builds on the existing proposal for
+[Forwarding Modules][]. This proposal inserts some additional detail to the
+final step of the process:
+
+[Forwarding Modules]: ../accepted/module-system.md#forwarding-modules
+
+> Previously: "Otherwise, add `member` to `module` with the name `name`."
+
+* Otherwise, if `rule` has a `withClause` `with`:
+
+ * If `file` contains an `@use` rule with the same target URL and another
+   `withClause`, throw an error.
+
+   > A module can only be configured from a single thread of `@forward`'s
+   > ending in an optional `@use`. Parallel configurations are not allowed,
+   > even if they come from the same source `file`.
+
+ * If `with` includes a `KeywordArgument` `agument` with the name `name`
+   (including `$`):
+
+   * If `member` does not have a `!default` flag, throw an error.
+
+     > We can only configure membes that are maked as default.
+
+   * Otherwise, let `value` be the value of `agument`, including the
+     `!default` flag if present.
+
+   * Add `member` to `module` with the name `name` and value `value`.
```

Comment:
@nex3 This is the sort of technical detail that I was getting hung up on. I'm not entirely clear if member has an available value at this point in the process, and if we can so simply change it, flags-and-all, as we pass things along. Doing it this way doesn't catch errors from external configurations, or demonstrate the configuration being applied when the forwarding module acts as an entrypoint…

(I was working from the "Executing Files" section in my first attempt, which is a bit more in-the-weeds…)

Reply:
In this case, I think it doesn't work, because according to [the definition of "member"](https://github.com/sass/sass/blob/master/accepted/module-system.md#member) each member has a definition; so if you're adding a new value, you have to create a new member with that value. That's not what you want to do here, because you want the forwarded member to refer to the same "slot" as the original; for example, if you write:

```scss
// upstream.scss
$var: original !default;
```

```scss
// midstream.scss
@forward "upstream" with ($var: overridden);
```

You want `upstream`'s value of `$var` to be `overridden` as well as `midstream`'s value (in fact you don't want them to be separate values at all).